R/merge_sce_list.R

```diff
@@ -118,6 +129,41 @@ merge_sce_list <- function(sce_list = list(),
   # Create the merged SCE from the processed list ------------------
   merged_sce <- do.call(cbind, sce_list)
 
+  # combine library metadata into a single data frame
+  metadata_list <- metadata(merged_sce)
+
+  # create a single list of library metadata
+  library_metadata <- metadata_list[names(metadata_list) == "library_metadata"]
```

Comment:
Can we do these operations based on the original `sce_list`? It somehow feels safer to me, because still I don't think I understand how the metadata works after `cbind()`...

So my thought would be that you would do something like this for the libraries:

```suggestion
# create a single list of library metadata lists
library_metadata  <- sce_list |> purrr::map("library_metadata")
```
(this is a really nice shorthand for pulling a single element out of each of a list of lists).

Reply:
Just to try and add some clarity into how the metadata looks after `cbind()`: 

```
$library_id
[1] "SCPCL000050"

$sample_id
[1] "SCPCS000053"

$salmon_version
[1] "1.9.0"

$reference_index
[1] "Homo_sapiens.GRCh38.104.spliced_intron.txome"
....

$library_id
[1] "SCPCL000051"

$sample_id
[1] "SCPCS000054"

$salmon_version
[1] "1.9.0"

$reference_index
[1] "Homo_sapiens.GRCh38.104.spliced_intron.txome"
...
```

So the lists are just concatenated together. Which is why I was pulling out anything with `library_metadata` as the name. Once we combine, we'll have a list named `library_metadata` for each object that was included in the merge.